elements/lisk-p2p/src/peer_directory/peer_book.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import { P2PDiscoveredPeerInfo, P2PPeerInfo } from '../p2p_types';
+import { NewPeerConfig, NewPeers } from './new_peers';
+import { TriedPeerConfig, TriedPeers } from './tried_peers';
+
+export interface PeerBookConfig {
+	readonly newPeerConfig?: NewPeerConfig;
+	readonly triedPeerConfig?: TriedPeerConfig;
+	readonly secret: number;
+}
+
+export class PeerBook {
+	private readonly _bannedPeers: ReadonlyArray<P2PDiscoveredPeerInfo>;
+	private readonly _newPeers: NewPeers;
+	private readonly _triedPeers: TriedPeers;
+	public constructor(peerBookConfig: PeerBookConfig) {
+		this._newPeers = new NewPeers(
+			peerBookConfig.newPeerConfig
+				? peerBookConfig.newPeerConfig
+				: { secret: peerBookConfig.secret },
+		);
+		this._triedPeers = new TriedPeers(
+			peerBookConfig.triedPeerConfig
+				? peerBookConfig.triedPeerConfig
+				: { secret: peerBookConfig.secret },
+		);
+		this._bannedPeers = [];
+	}
+
+	public get newPeers(): ReadonlyArray<P2PPeerInfo> {
+		return this._newPeers.getNewPeersList();
+	}
+
+	public get triedPeers(): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		return this._triedPeers.getTriedPeersList();
+	}
+
+	public get bannedPeers(): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		return this._bannedPeers;
+	}
+
+	public getAllPeers(): ReadonlyArray<P2PPeerInfo> {
+		return [...this.newPeers, ...this.triedPeers];
+	}
+
+	// If the peer is completed deleted then return true
```

Comment:
I dint understand this comment, can u explain to me what does it mean?

Reply:
The language that I wanted to write was `// If the peer is completely deleted from both the peer lists then return true`,
I want to return `true` when the peer is completely deleted for logging purpose at p2p class or we can use it for other advanced purposes like keeping track of deleted peers that are frequently found during the discovery again and again and maybe avoid connecting to them  next time.